### PR TITLE
provider/aws: Convert AWS ASG to upstream aws-sdk-go

### PR DESCRIPTION
**Pull request by catsby**
Converts AWS ASG resource and companion tag library.

### CODE REVIEW

File: builtin/providers/aws/config.go [hunk 1]
```diff
@@ -30,6 +31,7 @@ type AWSClient struct {
 	ec2conn         *ec2.EC2
 	elbconn         *elb.ELB
 	autoscalingconn *autoscaling.AutoScaling
+	asgconn         *awsASG.AutoScaling
```
Comment (catsby) [inline builtin/providers/aws/config.go]: we need a separate connection for the meantime, until Launch Config. is converted. LC is a non-trivial size resource so I didn't bundle it hear

File: builtin/providers/aws/resource_aws_autoscaling_group_test.go [hunk 2]
```diff
@@ -5,15 +5,16 @@ import (
 	"reflect"
 	"testing"
 
-	"github.com/hashicorp/aws-sdk-go/aws"
-	"github.com/hashicorp/aws-sdk-go/gen/autoscaling"
+	"github.com/awslabs/aws-sdk-go/aws"
+	"github.com/awslabs/aws-sdk-go/service/autoscaling"
+	hashiASG "github.com/hashicorp/aws-sdk-go/gen/autoscaling"
```
Comment (catsby) [inline builtin/providers/aws/resource_aws_autoscaling_group_test.go]: need to support `hashicorp` launch configuration

File: builtin/providers/aws/autoscaling_tags.go [hunk 3]
```diff
@@ -146,9 +146,9 @@ func autoscalingTagsToMap(ts []autoscaling.Tag) map[string]interface{} {
 }
 
 // autoscalingTagDescriptionsToMap turns the list of tags into a map.
-func autoscalingTagDescriptionsToMap(ts []autoscaling.TagDescription) map[string]map[string]interface{} {
+func autoscalingTagDescriptionsToMap(ts *[]*autoscaling.TagDescription) map[string]map[string]interface{} {
```
Comment (phinze) [inline builtin/providers/aws/autoscaling_tags.go]: `*[]*`

tee hee

File: builtin/providers/aws/config.go (hunk 1 above)
Comment (phinze) [inline builtin/providers/aws/config.go]: :+1: good call

**Comment by phinze** [comment]
LGTM

**Pull request merged**